plugins/inputs/jenkins/jenkins.go

```diff
@@ -0,0 +1,440 @@
+		if diskSpace, ok = diskSpaceMonitor.(map[string]interface{}); !ok {
+			return badFormatErr(url, diskSpaceMonitor, "map[string]interface{}", diskSpaceMonitorRoute)
+		}
+		if tags["disk_path"], ok = diskSpace["path"].(string); !ok {
+			return badFormatErr(url, diskSpace["path"], "string", diskSpaceMonitorRoute+".path")
+		}
+		if fields["disk_available"], ok = diskSpace["size"].(float64); !ok {
+			return badFormatErr(url, diskSpace["size"], "float64", diskSpaceMonitorRoute+".size")
+		}
+	}
+
+	if tempSpaceMonitor := info.MonitorData.Hudson_NodeMonitors_TemporarySpaceMonitor; tempSpaceMonitor != nil {
+		tempSpaceMonitorRoute := "hudson.node_monitors.TemporarySpaceMonitor"
+		var tempSpace map[string]interface{}
+		if tempSpace, ok = tempSpaceMonitor.(map[string]interface{}); !ok {
+			return badFormatErr(url, tempSpaceMonitor, "map[string]interface{}", tempSpaceMonitorRoute)
+		}
+		if tags["temp_path"], ok = tempSpace["path"].(string); !ok {
+			return badFormatErr(url, tempSpace["path"], "string", tempSpaceMonitorRoute+".path")
+		}
+		if fields["temp_available"], ok = tempSpace["size"].(float64); !ok {
+			return badFormatErr(url, tempSpace["size"], "float64", tempSpaceMonitorRoute+".size")
+		}
+	}
+
+	if swapSpaceMonitor := info.MonitorData.Hudson_NodeMonitors_SwapSpaceMonitor; swapSpaceMonitor != nil {
+		swapSpaceMonitorRouter := "hudson.node_monitors.SwapSpaceMonitor"
+		var swapSpace map[string]interface{}
+		if swapSpace, ok = swapSpaceMonitor.(map[string]interface{}); !ok {
+			return badFormatErr(url, swapSpaceMonitor, "map[string]interface{}", swapSpaceMonitorRouter)
+		}
+		if fields["swap_available"], ok = swapSpace["availableSwapSpace"].(float64); !ok {
+			return badFormatErr(url, swapSpace["availableSwapSpace"], "float64", swapSpaceMonitorRouter+".availableSwapSpace")
+		}
+		if fields["swap_total"], ok = swapSpace["totalSwapSpace"].(float64); !ok {
+			return badFormatErr(url, swapSpace["totalSwapSpace"], "float64", swapSpaceMonitorRouter+".totalSwapSpace")
+		}
+		if fields["memory_available"], ok = swapSpace["availablePhysicalMemory"].(float64); !ok {
+			return badFormatErr(url, swapSpace["availablePhysicalMemory"], "float64", swapSpaceMonitorRouter+".availablePhysicalMemory")
+		}
+		if fields["memory_total"], ok = swapSpace["totalPhysicalMemory"].(float64); !ok {
+			return badFormatErr(url, swapSpace["totalPhysicalMemory"], "float64", swapSpaceMonitorRouter+".totalPhysicalMemory")
+		}
+	}
+	acc.AddFields(measurementNode, fields, tags)
+
+	return nil
+}
+
+func (j *Jenkins) gatherNodesData(acc telegraf.Accumulator) {
+	var nodes []*gojenkins.Node
+	var err error
+	err = j.doGet(func() error {
+		nodes, err = j.instance.GetAllNodes()
+		return err
+	})
+
+	url := j.URL + "/computer/api/json"
+	// since gojenkins lib will never return error
+	// returns error for len(nodes) is 0
+	if err != nil || len(nodes) == 0 {
+		acc.AddError(fmt.Errorf("error retrieving nodes[%s]: %v", url, err))
```

Comment:
Hey, @danielnelson, thanks for the mention. I have a quick question for this one actually. I feel like if a jenkins service is running, it must have at least one node. Since gojenkins doesn't return the decoding error, a blank 200 page, will still be good. Do you think we should patch the forked gojenkins lib? or we just keep something like this?